tls/s2n_x509_validator.c

```diff
@@ -648,6 +648,32 @@ static S2N_RESULT s2n_x509_validator_disable_time_validation(struct s2n_connecti
     return S2N_RESULT_OK;
 }
 
+int no_op_verify_custom_crit_oids_cb(X509_STORE_CTX *ctx, X509 *x509, STACK_OF(ASN1_OBJECT) *oids)
+{
+    return 1;
+}
+
+static S2N_RESULT s2n_x509_validator_add_custom_extensions(struct s2n_x509_validator *validator, struct s2n_connection *conn)
+{
+#if S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID
+    if (conn->config->custom_crit_oids) {
```

Comment:
I think I'd prefer to swap the order of these branches. That way we can validate that we're in a state we expect rather than just proceeding. Like, `conn->config->custom_crit_oids` should never be true when `S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID` is false:
```
if (conn->config->custom_crit_oids) {
#if S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID
    ...
#else
    RESULT_BAIL(S2N_ERR_UNIMPLEMENTED);
#endif
}
```